pages/docs/couple-your-code/couple-your-code-gradient-data.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Step 9 – Gradient data 
+permalink: couple-your-code-gradient-data.html
+keywords: api, adapter, mapping, gradient
+summary: "So far, our mesh contains only data. This is sufficient for most of the numerical methods that preCICE offers. For nearest-neighbor-gradient mapping, however, preCICE also requires additional gradient data information. In this step, you learn how to add gradient data to the mesh."
+---
+
+{% version 2.4.0 %}
+This feature is available since version 2.4.0.
+{% endversion %}
+
+When using `nearest-neighbor-gradient` mapping, we require coupling data and additional gradient data. We have seen in [Step 3](couple-your-code-mesh-and-data-access.html) how to write data to the mesh.
+Now, we will learn how to write gradient data to the mesh.
+
+For this purpose, we use the following API methods:
+
+```cpp
+bool isDataGradientRequired(int dataID);
+
+void writeScalarGradientData (
+    int             dataID,
+    int             valueIndex,
+    const double*   gradientValues  );
+
+void writeBlockScalarGradientData (
+      int           dataID,
+      int           size,
+      const int*    valueIndices,
+      const double* gradientValues  );
+      
+void writeVectorGradientData (
+    int             dataID,
+    int             valueIndex,
+    const double*   gradientValues,
+    bool            rowsFirst = false );
+
+void writeBlockVectorGradientData (
+      int           dataID,
+      int           size,
+      const int*    valueIndices,
+      const double* gradientValues,
+      bool          rowsFirst = false );
+```
+
+* `isDataGradientRequired` returns a boolean, indicates if the data corresponding to the ID `dataID` has gradient data.
+* `writeScalarGradientData` writes scalar-valued gradient data, derived in each spatial dimension to the coupling data structure.
+* `ẁriteBlockScalarGradintData` writes multiple scalar gradient data at once, for performance reasons.
+* `writeVectorGradientData` writes vector-valued gradient data to the coupling data structure. The matrix is entered as a 1D-array of each component differentiated first. The parameter `rowsFirst` allows to write the values of the matrix, such as the components are differentiated in the spatial dimensions first.
```

Comment:
I know, of course, what you mean, but I have troubles understanding the last two sentences.